Make the app shows the top tracks, artists, and recently played tra…

usamaeltmsah opened the pull request:
Make the app show the top tracks, top artists, and recently played tracks for the Spotify user.

usamaeltmsah committed:
Make the app shows the top tracks, artists, and the recent played tracks for the Spotify user (Spotify Statistics)

Peter-Schorn commented:
I'm only going to merge the minimal amount of changes necessary. I'm not going to rename my entire project to "Topotify". You can't be serious.

Peter-Schorn commented on Topotify/Views/TabBarViews/TopArtistsView.swift:
```diff
@@ -0,0 +1,216 @@
+    
+    @EnvironmentObject var spotify: Spotify
+
+    @State private var topArtists: [Artist]
```
Just set the initial value of `topArtists` at the declaration site and get rid of the initializer.

usamaeltmsah commented:
Thanks, Peter, for your response. I got your point, and I will review the
code.

On Sat, Jul 23, 2022 at 11:20 PM Peter Schorn ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In Topotify/Views/TabBarViews/TopArtistsView.swift
> <https://github.com/Peter-Schorn/SpotifyAPIExampleApp/pull/4#discussion_r928166087>
> :
>
> > +//  Topotify
> +//
> +//  Created by Usama Fouad on 23/07/2022.
> +//
> +
> +import SwiftUI
> +import Combine
> +import SpotifyWebAPI
> +import SpotifyExampleContent
> +
> +
> +struct TopArtistsView: View {
> +
> +    @EnvironmentObject var spotify: Spotify
> +
> +    @State private var topArtists: [Artist]
>
> Just set the initial value at the declaration site and get rid of the
> initializer.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Peter-Schorn/SpotifyAPIExampleApp/pull/4#pullrequestreview-1048570185>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFSMAWNP5NGGQBQOVMB2GVDVVROZ5ANCNFSM54OIRISQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

usamaeltmsah committed:
Set the initial value of topArtists, recentlyPlayed and topSongs

Peter-Schorn commented on Topotify/Views/TabBarViews/TopArtistsView.swift:
```diff
@@ -0,0 +1,216 @@
+    
+    @EnvironmentObject var spotify: Spotify
+
+    @State private var topArtists = [Artist]()
```
Change this to `@State private var topArtists: [Artist] = []`

usamaeltmsah committed:
Update TopArtistsView.swift

Replace the @State private var topArtists = [Artist]() with @State private var topArtists: [Artist] = []
